File: src/main/java/chapter4/create/IntervalExample.java
package chapter4.create;

import common.CommonUtils;
import common.Log;
import io.reactivex.Observable;

import java.util.concurrent.TimeUnit;


public class IntervalExample {

    public void printNumbers() {
        CommonUtils.exampleStart();
        Observable<Long> source = Observable.interval(100L, TimeUnit.MILLISECONDS)
                .map(data -> (data + 1) * 100)
                .take(5);
        source.subscribe(Log::it);
        CommonUtils.sleep(1000);
    }

    public void noInitialDelay() {
        CommonUtils.exampleStart();
        Observable<Long> source = Observable.interval(0L, 100L, TimeUnit.MILLISECONDS)
                .map(val -> val + 100)
                .take(5);
        source.subscribe(Log::it);
        CommonUtils.sleep(1000);
    }

    public static void main(String[] args) {
        IntervalExample demo = new IntervalExample();
        demo.printNumbers();
        demo.noInitialDelay();
    }

}

File: src/main/kotlin/chapter4/create/IntervalExample_.kt
package chapter4.create

import common.CommonUtils_
import common.Log_
import io.reactivex.Observable
import java.util.concurrent.TimeUnit

class IntervalExample_ {

    fun printNumbers() {
        CommonUtils_.exampleStart()
        val source = Observable.interval(100L, TimeUnit.MILLISECONDS)
                .map { (it + 1) * 100 }
                .take(5)
        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(1000)
    }

    fun noInitialDelay() {
        CommonUtils_.exampleStart()
        val source = Observable.interval(0L, 100L, TimeUnit.MILLISECONDS)
                .map { it + 100 }
                .take(5)
        source.subscribe { Log_.it(it) }
        CommonUtils_.sleep(1000)
    }

}


fun main(args: Array<String>) {
    val demo = IntervalExample_()
    demo.printNumbers()
    demo.noInitialDelay()
}
